### DISCUSSION TITLE
Ability to Customize "Notices"

### DISCUSSION THREAD

**Discussion by ChuckJonas** [outline/outline#5159] [General]
The Notice (Info, Tip, Warn) is one of the most powerful ways to breakup big walls of text and make documentation more digestable.  

However, the current implementation feels very limited...  To me, the tip color comes across as a "Warning" as it shares the color [adopted by most UX libraries](https://ant.design/components/alert) and there is no "warning" or "success" notice.  

### Solutions:

**Lowest Effort**
- Add a "warn" & "success" notice
- Change the Tip color to something more neutral purple or orange.

**Moderate Effort** 
- Add a "warn" & "success" notice
- Add a setting to configure the notice colors

**Best Solution**
- Add a setting to add new "Notices"
- Add a setting to configure the notice colors & icon

**Comment by Limezy** [comment]
That would be indeed a great feature !

For those who are interested and who build Outline from sources instead of using the docker image, here is a working patch that I'm using for my own instance : https://github.com/Limezy/outline_ynh/blob/v0.68.0-anthropologiebiblique-comments/sources/patches/app-04-bible.patch. It adds a new notice type : in this particular case, it adds a Bible quote, in brown, but colors and icon should be very easy to change
<img width="771" alt="image" src="https://user-images.githubusercontent.com/24638389/230528460-4c035f45-664a-466a-830e-8699d1061d9b.png">

**Comment by tommoor** [comment]
Introducing a "Success" type makes a lot of sense and is a relatively small patch, as shown above. 

I noticed that the "Info" notice is using the accent color which is actually problematic as that can change to literally any color, that needs changing / fixing too.

**Comment by ChuckJonas** [comment]
Sorry, I mis-spoke above when I said `add a "warn" & "success" notice`, as "Warn" already exists.  

Short of having control to fully customize notices, I think in my ideal world, the notices would look like this:

- `success`
   - Color: green
   - Icon: ✅ 
- `tip`
   - Color: purple
   - Icon: 💡 
- `warning`
   - color: yellow
   - icon: ⚠️ 
- `error`
   - color: red
   - icon: ❌ 
   
   Though, I think @Limezy's scenario really shows the need for full customization.

**Reply by tommoor** [reply]
The names are just suggestions and aren't displayed on the end document, you can of course use the "Tip" notice as a warning if you prefer.

**Comment by kingfungdesign** [comment]
FWIW, I really like how Material for MKDocs handles it funtionally and aesthetically. Maybe a good reference.

https://squidfunk.github.io/mkdocs-material/reference/admonitions/
https://github.com/squidfunk/mkdocs-material
